planner/core/stats.go

```diff
@@ -150,12 +151,109 @@ func (ds *DataSource) DeriveStats(childStats []*property.StatsInfo) (*property.S
 			break
 		}
 	}
+	// consider the IndexMergePath. Now, we just generate `IndexMergePath` in DNF case.
+	if len(ds.pushedDownConds) > 0 && len(ds.possibleAccessPaths) > 2 && ds.ctx.GetSessionVars().EnableIndexMerge {
+		needConsiderIndexMerge := true
+		for i := 1; i < len(ds.possibleAccessPaths); i++ {
```

Comment:
Can we also consider the table access path? If there are access conditions on it, we can also make use of the benefit of index merge.

Reply:
can you give an example? i am confused about it.

Reply:
can you give an example? i am confused about it.

Reply:
For `create table t(a int primary key, b int, index idx(b))`, the `condition `a = 1 or b = 1` could also use index merge, since `a` is the handle of the table.